pkg/apis/openebs.io/v1alpha1/cas_volume.go

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright 2017 The OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CASVolumeKey is a typed string to represent cas volume related annotations'
+// or labels' keys
+//
+// Example - Below is a sample CASVolume that makes use of some of these
+// constants that are of type CASVolumeKey
+//
+//  kind: CASVolume
+//  apiVersion: v1alpha1
+//  metadata:
+//    name: jiva-cas-vol
+//    labels:
+//      # deprecated way to set capacity
+//      volumeprovisioner.mapi.openebs.io/storage-size: 2G
+//      k8s.io/storage-class: openebs-repaffinity-0.6.0
+//      k8s.io/namespace: default
+//      k8s.io/pvc: openebs-repaffinity-0.6.0
+//  # latest way to set capacity
+//  capacity: 2G
+type CASVolumeKey string
+
+const (
+	// CASTemplateCVK is the key to fetch name of CASTemplate custom resource
+	// to create a cas volume
+	CASTemplateCVK CASVolumeKey = "cas.openebs.io/template"
```

Comment:
following with the convention for read, list, delete, should this be called as create-template. This will help us tie in "upgrade-templates" or update-templates later on.

Reply:
yes... will change this.